src/olmo_core/nn/transformer/model.py

```diff
@@ -416,8 +430,12 @@ def _prepare_inputs(
         else:
             input_ids = move_to_device(input_ids, self.device)
             labels = move_to_device(labels, self.device)
-            all_block_kwargs["max_doc_len"] = max_doc_len
-            all_block_kwargs["cu_doc_lens"] = move_to_device(cu_doc_lens, self.device)
+
+            if max_doc_len is not None or cu_doc_lens is not None:
+                all_block_kwargs["max_doc_len"] = max_doc_len
+                all_block_kwargs["cu_doc_lens"] = move_to_device(cu_doc_lens, self.device)
+            elif cache_leftpad is not None:
+                all_block_kwargs["cache_leftpad"] = move_to_device(cache_leftpad, self.device)
```

Comment:
Are these fundamentally mutually exclusive, or is it just harder to support both? In any case, should we raise an exception if both are supplied?